lib/ansible/modules/windows/win_git.ps1

```diff
@@ -0,0 +1,421 @@
+            $Return.git_output = $git_output
+            
+            if ($LASTEXITCODE -eq 0) {
+               $result.changed = $true           
+               Set-Attr $result.win_git "msg" "Successfully cloned $repo into $dest."    
+            }                    
+     
+        }    
+        else {
+            Set-Attr $result.win_git "msg" "Failed to clone $repo into $dest. It is either not empty or does not exist"    
+        }
+
+        Set-Attr $result.win_git "return_code" $LASTEXITCODE
+        Set-Attr $result.win_git "git_output" $git_output
+    }
+    else {
+        $Return.rc = 0
+        $Return.git_output = $git_output        
+        Set-Attr $result.win_git "msg" "Check Mode - Skipping clone of $repo"
+    }
+
+    # Check if branch is the correct one
+    if ($version -ne "HEAD") {
+        Set-Location $dest; &git status --short --branch $version | Tee-Object -Variable branch_status | Out-Null
+        $branch_status = $branch_status.split("/")[1]
+        Set-Attr $result.win_git "branch_status" "$branch_status"
+
+            if ( $branch_status -ne "$version" ) {
+                Fail-Json $result "Branch $branch_status is not $branch"
+            }
+    }     
+
+    return $Return
+}
+
+function update {
+    # git clone command
+    [CmdletBinding()]
+    param()
+
+    Set-Attr $result.win_git "method" "pull"
+    [hashtable]$Return = @{}
+    $git_output = ""
+
+    # Build Arguments
+    $git_opts = @()
+    $git_opts += "--no-pager"
+    $git_opts += "pull"
+    $git_opts += $remote
+    
+    if ($version -ne "HEAD") {
+       $git_opts += $version
+    } 
+
+    Set-Attr $result.win_git "git_opts" $git_opts
+    #Only update if $dest does exist and not in check mode
+    if(-Not $check_mode) {
+        # move into correct branch before pull
+        checkout
+        # perform git pull
+        Set-Location $dest; &git $git_opts | Tee-Object -Variable git_output | Out-Null
+        $Return.rc = $LASTEXITCODE
+        $Return.git_output = $git_output
+        if ($LASTEXITCODE -eq 0) {
+            Set-Attr $result "changed" $true
+            Set-Attr $result.win_git "msg" "Successfully updated $repo to $version."
+        }  
+        #TODO: handle correct status change when using update        
+        Set-Attr $result.win_git "return_code" $LASTEXITCODE
+        Set-Attr $result.win_git "git_output" $git_output
+    } else {
+        $Return.rc = 0
+        $Return.git_output =  $git_output
+        Set-Attr $result.win_git "msg" "Check Mode - Skipping update of $repo"
+    }
+
+    return $Return
+}
+
+
+if ($repo -eq ($null -or "")) {
+    Fail-Json $result "Repository cannot be empty or `$null"
+}
+
+# set the default dest under user profile
+if ($dest -eq $null){
+    $repo_name = $repo.split("/")[-1]
+    $dest="$env:USERPROFILE\$repo_name"     
+} 
+
+if (-Not (Test-Path($dest))) {
+    New-Item -ItemType directory -Path $dest      
+ }
+
+
+Set-Attr $result.win_git "repo" $repo
```

Comment:
I think this and L382 -> L388 can now be written without using 'Set-Attr' for example like this:

$result.win_git "repo" $repo

If I recall correctly ansible is slowly moving away from putting results into a subkey named after the module so (i.e not having win_git level in the results), and also not including any unmodified invocation parameters in the results (since you can get invocation parameters by running playbooks with -vvv), so it would be good to simplify these so the parameters passed to the module are not returned in the results.

Reply:
@jhawkesworth removed the subkey named output and non modified param